src/helper/service.hpp

```diff
@@ -57,6 +59,7 @@ class service {
     service_identifier id_;
     std::shared_ptr<engine> engine_;
     remote_config::client::ptr rc_client_;
+    std::shared_ptr<service_config> rc_service_;
```

Comment:
It's not really an rc_service it's the configuration of the service, the name should reflect that.